python/lsst/ts/salobj/sal_info.py

```diff
@@ -747,15 +747,14 @@ async def _read_loop(self, loop: asyncio.AbstractEventLoop) -> None:
         try:
             with concurrent.futures.ThreadPoolExecutor(max_workers=1) as pool:
                 while self.isopen:
-                    reader, data_list = await loop.run_in_executor(
-                        pool, self._blocking_read
-                    )
+                    reads = await loop.run_in_executor(pool, self._blocking_read)
                     if not self.isopen:
                         break
-                    if not data_list or reader is None:
-                        continue
-                    reader.dds_queue_length_checker.check_nitems(len(data_list))
-                    reader._queue_data(data_list)
+                    for reader, data_list in reads:
+                        if not data_list:
+                            continue
```

Comment:
Please delete this `if not data_list:` block. With a suggested change below the list will never be empty. And even if something does go wrong, the code will still be safe -- this is only intended as a no-longer-needed small optimization.

Reply:
Please make this change (or if you don't want to, please explain why -- you marked this resolved without changing it).